### PR TITLE
refactor: refine cycle reference error messages and add unit test cases.

### PR DESCRIPTION
**Pull request by Peefy**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

re #101 

#### 2. What is the scope of this PR (e.g. component or file name):

* kclvm/sema/src/resolver/global.rs
* kclvm/sema/src/resolver/import.rs

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

The test case `test_resolve_program_cycle_reference_fail` in kclvm/sema/src/resolver/tests.rs

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by zong-zhe — DISMISSED** [review]
LGTM

File: kclvm/sema/src/resolver/tests.rs [hunk 1]
```diff
@@ -79,3 +79,24 @@ fn test_resolve_program_fail() {
...
+#[test]
+fn test_resolve_program_cycle_reference_fail() {
+    let mut program = load_program(
+        &["./src/resolver/test_fail_data/cycle_reference/file1.k"],
+        None,
+    )
+    .unwrap();
+    let scope = resolve_program(&mut program);
+    let err_messages = [
+        "There is a circular import reference between module file1 and file2",
```
Comment (chai2010) [inline kclvm/sema/src/resolver/tests.rs:92]: how to report circle import error for file1->file2->file3->file1?

Comment (Peefy) [reply kclvm/sema/src/resolver/tests.rs:92]: > how to report circle import error for file1->file2->file3->file1?

There is a circular import reference between module file1 and file3

**Review by chai2010 — DISMISSED** [review]
LGTM

File: kclvm/sema/src/ty/context.rs [hunk 2]
```diff
@@ -52,6 +55,27 @@ impl TypeContext {
         is_cyclic_directed(&self.dep_graph)
     }
 
+    /// Add dependencies and return the dependency graph is cyclic.
+    pub fn add_dependencies(&mut self, from: &str, to: &str) -> bool {
```
Comment (chai2010) [inline kclvm/sema/src/ty/context.rs:59]: the bool result may lead to ambiguity: add succeed or failed?

File: kclvm/sema/src/ty/context.rs [hunk 3]
```diff
@@ -52,6 +55,27 @@ impl TypeContext {
...
+        let from_idx = self.get_node_index(from);
+        let to_idx = self.get_node_index(to);
+        self.dep_graph.add_edge(from_idx, to_idx, ());
+        self.is_cyclic()
+    }
+
+    /// Get the node index from the node index map or insert it into the dependency graph.
+    #[inline]
+    fn get_node_index(&mut self, name: &str) -> NodeIndex {
```
Comment (chai2010) [inline kclvm/sema/src/ty/context.rs:68]: get call will change the self, get or `midify` ...?

File: kclvm/sema/src/ty/context.rs (hunk 2 above)
Comment (Peefy) [reply kclvm/sema/src/ty/context.rs:59]: > the bool result may lead to ambiguity: add succeed or failed?

I will fix it on another PR.

File: kclvm/sema/src/ty/context.rs (hunk 3 above)
Comment (Peefy) [reply kclvm/sema/src/ty/context.rs:68]: > get call will change the self, use get or `get_or_insert_...` name?

I will fix it on another PR.

File: kclvm/sema/src/ty/context.rs (hunk 2 above)
Comment (Peefy) [reply kclvm/sema/src/ty/context.rs]: > the bool result may lead to ambiguity: add succeed or failed?

fixed.

File: kclvm/sema/src/ty/context.rs (hunk 3 above)
Comment (Peefy) [reply kclvm/sema/src/ty/context.rs]: > get call will change the self, use get or `get_or_insert_...` name?

fixed.

File: kclvm/sema/src/ty/context.rs [hunk 4]
```diff
@@ -52,6 +55,27 @@ impl TypeContext {
         is_cyclic_directed(&self.dep_graph)
     }
 
+    /// Add dependencies and return the dependency graph is cyclic.
+    pub fn add_dependencies_and_is_cyclic(&mut self, from: &str, to: &str) -> bool {
```
Comment (chai2010) [inline kclvm/sema/src/ty/context.rs:59]: split `add_dependencies_and_is_cyclic` to `add_dependencies` and `self.is_cyclic()`

**Review by chai2010 — APPROVED** [review]
LGTM

**Review by chai2010 — APPROVED** [review]

**Review by He1pa — APPROVED** [review]
LGTM

**Pull request merged**